samples/csharp_dotnetcore/22.qnamaker-App_insights/Bots/QnABot.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.AI.QnA;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class QnABot : ActivityHandler
+    {
+        private readonly IConfiguration _configuration;
+        private readonly ILogger<QnABot> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IBotTelemetryClient _telemetryClient;
+
+        public QnABot(IConfiguration configuration, ILogger<QnABot> logger, IHttpClientFactory httpClientFactory, IBotTelemetryClient telemetryClient)
+        {
+            _configuration = configuration;
+            _logger = logger;
+            _httpClientFactory = httpClientFactory;
+            _telemetryClient = telemetryClient;
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var httpClient = _httpClientFactory.CreateClient();
+
+            var qnaMaker = new MyQnAMaker(new QnAMakerEndpoint
```

Comment:
Can we inject this through DI as well?

Reply:
Eyal in his email Sent on Monday, August 26, 2019 12:58 PM, Suggested leaving it as is.  If you do not agree, we should hash it out.